--- a/remotes/vmm/kvm/migrate
+++ b/remotes/vmm/kvm/migrate
@@ -22,6 +22,21 @@ source $(dirname $0)/../../scripts_common.sh
 deploy_id=$1
 dest_host=$2
 
+memEntry="$(virsh --connect $LIBVIRT_URI dumpxml $deploy_id | grep "<memory ")"
+memEntry="${memEntry#*>}"
+memEntry="${memEntry%<*}"
+if [ -n "$memEntry" ] && [ -z "${memEntry//[[:digit:]]/}" ]; then
+    memGB=$((memEntry/1024**2))
+    if [ $memGB -gt 0 ] ; then
+        MIGRATE_TIMEOUT=$((memGB*2))
+    fi
+fi
+if [ -n "$MIGRATE_OPTIONS" ]; then
+    MIGRATE_OPTIONS+=" --timeout-postcopy --timeout=${MIGRATE_TIMEOUT:-600}"
+    log "MIGRATE_OPTIONS: $MIGRATE_OPTIONS"
+fi
+logger -t vmm_sp_migrate -- "$deploy_id memory=$memEntry $MIGRATE_OPTIONS"
+
 # migration can't be done with domain snapshots, drop them first
 snaps=$(monitor_and_log \
    "virsh --connect $LIBVIRT_URI snapshot-list $deploy_id --name 2>/dev/null" \
