diff -urN x.orig/var/lib/one/remotes/im/kvm-probes.d/monitor_ds.sh x/var/lib/one/remotes/im/kvm-probes.d/monitor_ds.sh
--- remotes/im/kvm-probes.d/monitor_ds.sh	2015-09-24 15:34:36.000000000 +0300
+++ remotes/im/kvm-probes.d/monitor_ds.sh	2015-09-29 15:54:07.559284564 +0300
@@ -48,6 +48,8 @@
         LVM_STATUS=255
     fi
 
+    [ -f ../../datastore/storpool/monitor_ds.sh ] && source ../../datastore/storpool/monitor_ds.sh
+
     echo "DS = ["
     echo "  ID = $ds,"
 
