diff -ur remotes.orig/vnm/sg_driver.rb remotes/vnm/sg_driver.rb
--- remotes.orig/vnm/sg_driver.rb	2024-11-06 19:34:12.000000000 +0200
+++ remotes/vnm/sg_driver.rb	2024-11-07 16:12:54.924814848 +0200
@@ -125,7 +125,7 @@
                         sg.run!
                     rescue StandardError => e
                         unlock
-                        deactivate(do_all)
+                        SGIPTables.nic_post(@vm, nic)
                         raise e
                     end
                 end
