--- remotes/tm/ssh/monitor_ds.orig	2018-06-03 22:52:04.000000000 +0000
+++ remotes/tm/ssh/monitor_ds	2018-06-09 11:40:55.393568543 +0000
@@ -29,6 +29,19 @@
     echo -n "VM=[ID=$vm,POLL=\""
 
     for disk in $disks; do
+        if [ -f "${vmdir}/${disk}.monitor" ]; then
+            DRIVER="$(<"${vmdir}/${disk}.monitor")"
+            for tm_path in "$PWD/remotes/tm" "/var/tmp/one/tm"; do
+                if [ -x "$tm_path/$DRIVER/monitor_disk" ]; then
+                    DRIVER_RUN="$tm_path/$DRIVER/monitor_disk"
+                    break
+                fi
+            done
+            if [ -n "$DRIVER_RUN" ]; then
+                "$DRIVER_RUN" "${vmdir}/${disk}" "${DS_ID}" "$vm" "$DRIVER"
+                continue
+            fi
+        fi
         disk_id="$(echo "$disk" | cut -d. -f2)"
         disk_size="$(du -mL "${vmdir}/${disk}" 2>/dev/null | awk '{print $1}')"
         snap_dir="${vmdir}/${disk}.snap"
