--- remotes/vmm/kvm/restore.orig	2017-06-25 23:04:09.698231380 +0300
+++ remotes/vmm/kvm/restore	2017-06-25 23:05:52.546656416 +0300
@@ -80,3 +80,14 @@
 
 rm "$FILE"
 rm "$FILE_XML"
+
+if [ -t 0 ]; then
+    exit 0
+fi
+
+# If there is a specific post hook for this TM_MAD call it:
+RESTORE_TM_FILE="${DRIVER_PATH}/restore.${TM_MAD}-post"
+
+if [ -x "$RESTORE_TM_FILE" ]; then
+    echo "$DRV_MESSAGE" | $RESTORE_TM_FILE "$@"
+fi
