--- app/tabs/datastores-tab/form-panels/create/wizard.hbs	2016-10-17 11:09:04.000000000 +0000
+++ app/tabs/datastores-tab/form-panels/create/wizard.hbs	2016-11-08 14:31:27.610021610 +0000
@@ -33,6 +33,7 @@
         <option ds="fs" tm="lvm" value="lvm">{{tr "LVM"}}</option>
         <option ds="dev" tm="dev" value="raw">{{tr "Raw Device Mapping"}}</option>
         <option ds="iscsi" tm="iscsi" value="iscsi_libvirt">{{tr "iSCSI - Libvirt initiator "}}</option>
+        <option value="storpool">{{tr "StorPool"}}</option>
         <option value="custom">{{tr "Custom"}}</option>
       </select>
     </div>
@@ -53,6 +54,7 @@
             <option value="dev">{{tr "Devices"}}</option>
             <option value="iscsi_libvirt">{{tr "iSCSI/Libvirt"}}</option>
             <option value="vcenter">{{tr "vCenter"}}</option>
+            <option value="storpool">{{tr "StorPool"}}</option>
             <option value="custom">{{tr "Custom"}}</option>
           </select>
           <div>
@@ -71,6 +73,7 @@
             <option value="dev">{{tr "Devices"}}</option>
             <option value="iscsi_libvirt">{{tr "iSCSI/Libvirt"}}</option>
             <option value="vcenter">{{tr "vCenter"}}</option>
+            <option value="storpool">{{tr "StorPool"}}</option>
             <option value="custom">{{tr "Custom"}}</option>
           </select>
           <div>
@@ -262,5 +265,68 @@
         <div id="vcenter_cluster_wrapper"></div>
       </label>
     </div>
+    <div class="medium-6 columns">
+      <label for="sp_replication">
+        {{tr "StorPool Replication"}}
+        <span class="tip">
+          {{tr "StorPool can keep between 1 and 3 copies of data spread on all servers in the cluster."}}
+        </span>
+      </label>
+      <input type="text" name="sp_replication" id="sp_replication" />
+    </div>
+    <div class="medium-6 columns">
+      <label for="sp_placeall">
+        {{tr "StorPool PlaceAll"}}
+        <span class="tip">
+          {{tr "PlacementGroup to allocate data on. PlacementGroup must exists in the StorPool cluster. Check with 'storpool placementGroup list'."}}
+        </span>
+      </label>
+      <input type="text" name="sp_placeall" id="sp_placeall" />
+    </div>
+    <div class="medium-6 columns">
+      <label for="sp_placetail">
+        {{tr "StorPool PlaceTail"}}
+        <span class="tip">
+          {{tr "PlacementGroup to keep one copy on. This copy (the tail) is used for reads. PlacementGroup must exist the StorPool cluster. Check with 'storpool placementGroup list'."}}
+        </span>
+      </label>
+      <input type="text" name="sp_placetail" id="sp_placetail" />
+    </div>
+    <div class="medium-6 columns">
+      <label for="sp_api_http_host">
+        {{tr "StorPool API  host"}}
+        <span class="tip">
+          {{tr "Used when multiple clusters are managed by one frontend. Set to the IP of the desired cluster API interface"}}
+        </span>
+      </label>
+      <input type="text" name="sp_api_http_host" id="sp_api_http_host" />
+    </div>
+    <div class="medium-6 columns">
+      <label for="sp_api_http_port">
+        {{tr "StorPool API port"}}
+        <span class="tip">
+          {{tr "Used when multiple clusters are managed by one frontend. Set to the port where the desired cluster API interface is listening"}}
+        </span>
+      </label>
+      <input type="text" name="sp_api_http_port" id="sp_api_http_port" />
+    </div>
+    <div class="medium-6 columns">
+      <label for="sp_auth_token">
+        {{tr "StorPool API auth token"}}
+        <span class="tip">
+          {{tr "Used when multiple clusters are managed by one frontend. Set to the API token  for the desired cluster API"}}
+        </span>
+      </label>
+      <input type="text" name="sp_auth_token" id="sp_auth_token" />
+    </div>
+    <div class="medium-6 columns">
+      <label for="sp_system">
+        {{tr "StorPool system"}}
+        <span class="tip">
+          {{tr "Used when SYSTEM datastore type is selected. Set to 'ssh' if the datastore is not on shared fs"}}
+        </span>
+      </label>
+      <input type="text" name="sp_system" id="sp_system" />
+    </div>
   </div>
 </form>
--- app/tabs/datastores-tab/form-panels/create.js	2016-10-17 11:09:04.000000000 +0000
+++ app/tabs/datastores-tab/form-panels/create.js	2016-11-08 14:32:40.129764165 +0000
@@ -205,6 +205,11 @@
           _selectISCSI(dialog);
           break;
         
+        case "storpool":
+          $('input#file_ds_type', dialog).attr('disabled', 'disabled');
+          _selectStorPool(dialog);
+          break;
+        
         case 'custom':
           _selectCustom(dialog);
           break;
@@ -248,6 +253,14 @@
     var iscsi_user      = $('#iscsi_user', dialog).val();
     var iscsi_usage     = $('#iscsi_usage', dialog).val();
     var vcenter_cluster = $('#vcenter_cluster', dialog).val();
+    var sp_replication  = $('#sp_replication', dialog).val();
+    var sp_placeall     = $('#sp_placeall', dialog).val();
+    var sp_placetail    = $('#sp_placetail', dialog).val();
+    var sp_replication  = $('#sp_replication', dialog).val();
+    var sp_api_http_host = $('#sp_api_http_host', dialog).val();
+    var sp_api_http_port = $('#sp_api_http_port', dialog).val();
+    var sp_auth_token   = $('#sp_auth_token', dialog).val();
+    var sp_system       = $('#sp_system', dialog).val();
 
     var ds_obj = {
       "datastore" : {
@@ -330,6 +343,31 @@
     if (vcenter_cluster)
         ds_obj.datastore.vcenter_cluster = vcenter_cluster;
 
+    if (sp_replication)
+        ds_obj.datastore.sp_replication = sp_replication;
+
+    if (sp_placeall)
+        ds_obj.datastore.sp_placeall = sp_placeall;
+
+    if (sp_placetail)
+        ds_obj.datastore.sp_placetail = sp_placetail;
+
+    if (sp_api_http_host)
+        ds_obj.datastore.sp_api_http_host = sp_api_http_host;
+
+    if (sp_api_http_port)
+        ds_obj.datastore.sp_api_http_port = sp_api_http_port;
+
+    if (sp_auth_token)
+        ds_obj.datastore.sp_auth_token = sp_auth_token;
+
+    if (ds_type == "SYSTEM_DS") {
+        if (sp_system)
+            ds_obj.datastore.sp_system = sp_system;
+        if (tm_mad == "storpool")
+            ds_obj.datastore.disk_type = "file";
+    }
+
     Sunstone.runAction("Datastore.create", ds_obj);
     return false;
   }
@@ -383,6 +421,14 @@
 
     $(".disk_type_wrapper", dialog).hide();
 
+    $('label[for="sp_replication"],input#sp_replication', dialog).parent().hide();
+    $('label[for="sp_placeall"],input#sp_placeall', dialog).parent().hide();
+    $('label[for="sp_placetail"],input#sp_placetail', dialog).parent().hide();
+    $('label[for="sp_api_http_host"],input#sp_api_http_host', dialog).parent().hide();
+    $('label[for="sp_api_http_port"],input#sp_api_http_port', dialog).parent().hide();
+    $('label[for="sp_auth_token"],input#sp_auth_token', dialog).parent().hide();
+    $('label[for="sp_system"],input#sp_system', dialog).parent().hide();
+
     _resetAll(dialog);
   }
 
@@ -411,6 +457,14 @@
 
     $(".disk_type_wrapper", dialog).show();
 
+    $('label[for="sp_replication"],input#sp_replication', dialog).parent().show();
+    $('label[for="sp_placeall"],input#sp_placeall', dialog).parent().show();
+    $('label[for="sp_placetail"],input#sp_placetail', dialog).parent().show();
+    $('label[for="sp_api_http_host"],input#sp_api_http_host', dialog).parent().show();
+    $('label[for="sp_api_http_port"],input#sp_api_http_port', dialog).parent().show();
+    $('label[for="sp_auth_token"],input#sp_auth_token', dialog).parent().show();
+    $('label[for="sp_system"],input#sp_system', dialog).parent().show();
+
     _resetAll(dialog);
   }
 
@@ -503,6 +557,33 @@
     $('label[for="vcenter_cluster"],div#vcenter_cluster_wrapper', dialog).parent().fadeIn();
   }
 
+  function _selectStorPool(dialog) {
+    $('select#ds_mad', dialog).val('storpool');
+    $('select#ds_mad', dialog).attr('disabled', 'disabled');
+    $('select#tm_mad', dialog).val('storpool');
+    $('select#tm_mad', dialog).attr('disabled', 'disabled');
+    $('input#image_ds_type', dialog).attr('checked', 'true');
+    $('input#file_ds_type', dialog).attr('disabled', 'disabled');
+    $('label[for="sp_replication"],input#sp_replication', dialog).parent().fadeIn();
+    $('label[for="sp_placeall"],input#sp_placeall', dialog).parent().fadeIn();
+    $('label[for="sp_placetail"],input#sp_placetail', dialog).parent().fadeIn();
+    $('label[for="bridge_list"],input#bridge_list', dialog).parent().fadeIn();
+    $('label[for="sp_api_http_host"],input#sp_api_http_host', dialog).parent().fadeIn();
+    $('label[for="sp_api_http_port"],input#sp_api_http_port', dialog).parent().fadeIn();
+    $('label[for="sp_auth_token"],input#sp_auth_token', dialog).parent().fadeIn();
+    $('label[for="sp_system"],input#sp_system', dialog).parent().fadeIn();
+    $('label[for="staging_dir"],input#staging_dir', dialog).parent().fadeIn();
+    $('label[for="limit_transfer_bw"],input#limit_transfer_bw', dialog).parent().fadeIn();
+    $('label[for="no_decompress"],input#no_decompress', dialog).parent().fadeIn();
+    $('label[for="datastore_capacity_check"],input#datastore_capacity_check', dialog).parent().fadeIn();
+    $('select#disk_type', dialog).val('block');
+    $('select#disk_type', dialog).attr('disabled', 'disabled');
+    $('input#safe_dirs', dialog).removeAttr('disabled');
+    $('input#base_path', dialog).removeAttr('disabled');
+    $('input#limit_mb', dialog).removeAttr('disabled');
+    $('input#restricted_dirs', dialog).removeAttr('disabled');
+  }
+
   function _selectCustom(dialog) {
     _showAll(dialog);
 
