--- app/tabs/vms-tab/dialogs/revert.js
+++ app/tabs/vms-tab/dialogs/revert.js
@@ -48,6 +48,7 @@ define(function(require) {
   Dialog.prototype.onShow = _onShow;
   Dialog.prototype.setup = _setup;
   Dialog.prototype.setElement = _setElement;
+  Dialog.prototype.setSnapshot_id = _setSnapshot_id;
 
   return Dialog;
 
@@ -69,8 +70,7 @@ define(function(require) {
 
     $('#' + DIALOG_ID + 'Form', context).submit(function() {
 
-      var snapshot_id = $(this).parents('tr').attr('snapshot_id');
-      Sunstone.runAction('VM.snapshot_revert', that.element.ID,  {"snapshot_id": snapshot_id});
+      Sunstone.runAction('VM.snapshot_revert', that.element.ID,  {"snapshot_id": that.snapshot_id});
 
       Sunstone.getDialog(DIALOG_ID).hide();
       Sunstone.getDialog(DIALOG_ID).reset();
@@ -89,4 +89,9 @@ define(function(require) {
     this.element = element
   }
 
+
+  function _setSnapshot_id(snapshot_id) {
+    this.snapshot_id = snapshot_id
+  }
+
 });
--- app/tabs/vms-tab/panels/snapshots.js
+++ app/tabs/vms-tab/panels/snapshots.js
@@ -166,8 +166,10 @@ define(function(require) {
     if (Config.isTabActionEnabled("vms-tab", "VM.snapshot_revert")) {
       context.off('click', '.snapshot_revert');
       context.on('click', '.snapshot_revert', function() {
+        var snapshot_id = $(this).parents('tr').attr('snapshot_id');
         var dialog = Sunstone.getDialog(REVERT_DIALOG_ID);
         dialog.setElement(that.element);
+        dialog.setSnapshot_id(snapshot_id);
         dialog.show();
         return false;
       });
