diff -urN remotes.orig/vmm/kvm/attach_disk remotes/vmm/kvm/attach_disk
--- remotes.orig/vmm/kvm/attach_disk	2018-01-23 18:16:23.000000000 +0000
+++ remotes/vmm/kvm/attach_disk	2018-01-31 08:04:07.379207903 +0000
@@ -39,6 +39,10 @@
     DISCARD="$DEFAULT_ATTACH_DISCARD"
 [ -n "$DISCARD" ] && DISCARD="discard='$DISCARD'"
 
+[ -z "$IO" ] && [ -n "$DEFAULT_ATTACH_IO" ] && \
+    IO="$DEFAULT_ATTACH_IO"
+[ -n "$IO" ] && IO="io='$IO'"
+
 if [ "$READONLY" = "YES" ]; then
     READONLY="<readonly/>"
 else
@@ -47,7 +51,7 @@
 
 cat <<EOF > $ATTACH_FILE
 <disk type='$TYPE_XML' device='$DEVICE'>
-  <driver name='qemu' type='$DRIVER' $CACHE $DISCARD/>
+  <driver name='qemu' type='$DRIVER' $CACHE $DISCARD $IO/>
   <source $TYPE_SOURCE='$SOURCE' $SOURCE_ARGS>
   $SOURCE_HOST
   </source>
