diff -urN remotes.orig/vmm/kvm/kvmrc remotes/vmm/kvm/kvmrc
--- remotes.orig/vmm/kvm/kvmrc	2018-01-23 18:16:23.000000000 +0000
+++ remotes/vmm/kvm/kvmrc	2018-01-31 08:04:46.281529049 +0000
@@ -46,6 +46,11 @@
 # is "unmap" (or "on").
 #DEFAULT_ATTACH_DISCARD=unmap
 
+# This parameter will set default IO policy for new attached disks. The default
+# value depends on the version of qemu-kvm which is used. Possible values are
+# 'threads' or 'native'. Uncomment only one of the variants.
+#DEFAULT_ATTACH_IO=native
+
 # This parameters will set the default NIC model and filter for
 # the hot-attached interfaces.
 #DEFAULT_ATTACH_NIC_MODEL=virtio
