--- js/plugins/datastores-tab.js.orig	2015-03-13 10:40:31.129223366 +0200
+++ js/plugins/datastores-tab.js	2015-03-13 11:02:47.409756857 +0200
@@ -47,6 +47,7 @@
               <option value="ceph">' + tr("Ceph") + '</option>\
               <option value="gluster">' + tr("Gluster") + '</option>\
               <option value="dev">' + tr("Devices") + '</option>\
+              <option value="storpool">' + tr("StorPool") + '</option>\
               <option value="custom">' + tr("Custom") + '</option>\
             </select>\
           </div>\
@@ -81,6 +82,7 @@
                   <option value="vmfs">' + tr("VMFS") + '</option>\
                   <option value="ceph">' + tr("Ceph") + '</option>\
                   <option value="dev">' + tr("Devices") + '</option>\
+                  <option value="storpool">' + tr("StorPool") + '</option>\
                   <option value="custom">' + tr("Custom") + '</option>\
                 </select>\
                 <div>\
@@ -99,6 +101,7 @@
                   <option value="vmfs">' + tr("VMFS") + '</option>\
                   <option value="ceph">' + tr("Ceph") + '</option>\
                   <option value="dev">' + tr("Devices") + '</option>\
+                  <option value="storpool">' + tr("StorPool") + '</option>\
                   <option value="custom">' + tr("Custom") + '</option>\
                 </select>\
                 <div>\
@@ -224,6 +227,24 @@
               '</label>\
               <input type="text" name="rbd_format" id="rbd_format" />\
           </div>\
+          <div class="large-6 columns">\
+              <label for="sp_replication">' + tr("StorPool Replication") +
+                '<span class="tip">'+tr("StorPool can keep between 1 and 3 copies of data spread on all servers in the cluster.")+'</span>'+
+              '</label>\
+              <input type="text" name="sp_replication" id="sp_replication" />\
+          </div>\
+          <div class="large-6 columns">\
+              <label for="sp_placeall">' + tr("StorPool PlaceAll") +
+                '<span class="tip">'+tr("PlacementGroup to allocate data on. PlacementGroup must exists in the StorPool cluster. Check with 'storpool placementGroup list'.")+'</span>'+
+              '</label>\
+              <input type="text" name="sp_placeall" id="sp_placeall" />\
+          </div>\
+          <div class="large-6 columns">\
+              <label for="sp_placetail">' + tr("StorPool PlaceTail") +
+                '<span class="tip">'+tr("PlacementGroup to keep one copy on. This copy (the tail) is used for reads. PlacementGroup must exist the StorPool cluster. Check with 'storpool placementGroup list'.")+'</span>'+
+              '</label>\
+              <input type="text" name="sp_placetail" id="sp_placetail" />\
+          </div>\
         </div>\
         <div class="reveal-footer">\
           <div class="form_buttons">\
@@ -866,6 +887,9 @@
     $('label[for="rbd_format"],input#rbd_format',context).parent().hide();
     $('label[for="staging_dir"],input#staging_dir',context).parent().hide();
     $('label[for="limit_transfer_bw"],input#limit_transfer_bw',context).parent().hide();
+    $('label[for="sp_replication"],input#sp_replication',context).parent().hide();
+    $('label[for="sp_placeall"],input#sp_placeall',context).parent().hide();
+    $('label[for="sp_placetail"],input#sp_placetail',context).parent().hide();
     $('label[for="no_decompress"],input#no_decompress',context).parent().hide();
     $('select#ds_mad').removeAttr('disabled');
     $('select#tm_mad').removeAttr('disabled');
@@ -938,6 +962,9 @@
           case 'dev':
             select_devices();
             break;
+          case 'storpool':
+            select_storpool();
+            break;
           case 'custom':
             select_custom();
             break;
@@ -976,6 +1003,9 @@
         var rbd_format      = $('#rbd_format',context).val();
         var staging_dir     = $('#staging_dir',context).val();
 
+        var sp_replication  = $('#sp_replication',context).val();
+        var sp_placeall     = $('#sp_placeall',context).val();
+        var sp_placetail    = $('#sp_placetail',context).val();
 
         if (!name){
             notifyError("Please provide a name");
@@ -1051,6 +1081,15 @@
         if (staging_dir)
             ds_obj.datastore.staging_dir = staging_dir;
 
+        if (sp_replication)
+            ds_obj.datastore.sp_replication = sp_replication;
+
+        if (sp_placeall)
+            ds_obj.datastore.sp_placeall = sp_placeall;
+
+        if (sp_placetail)
+            ds_obj.datastore.sp_placetail = sp_placetail;
+
         Sunstone.runAction("Datastore.create",ds_obj);
         return false;
     });
@@ -1251,6 +1290,29 @@
     $('input#restricted_dirs').attr('disabled', 'disabled');
 }
 
+function select_storpool(){
+    $('select#ds_mad').val('storpool');
+    $('select#ds_mad').attr('disabled', 'disabled');
+    $('select#tm_mad').val('storpool');
+    $('select#tm_mad').attr('disabled', 'disabled');
+    $('input#image_ds_type').attr('checked', 'true');
+    $('input[name=ds_type]').attr('disabled', 'disabled');
+    $('label[for="sp_replication"],input#sp_replication').parent().fadeIn();
+    $('label[for="sp_placeall"],input#sp_placeall').parent().fadeIn();
+    $('label[for="sp_placetail"],input#sp_placetail').parent().fadeIn();
+    $('label[for="bridge_list"],input#bridge_list').parent().fadeIn();
+    $('label[for="staging_dir"],input#staging_dir').parent().fadeIn();
+    $('label[for="limit_transfer_bw"],input#limit_transfer_bw').parent().fadeIn();
+    $('label[for="no_decompress"],input#no_decompress').parent().fadeIn();
+    $('label[for="datastore_capacity_check"],input#datastore_capacity_check').parent().fadeIn();
+    $('select#disk_type').val('block');
+    $('select#disk_type').attr('disabled', 'disabled');
+    $('input#safe_dirs').removeAttr('disabled');
+    $('input#base_path').removeAttr('disabled');
+    $('input#limit_mb').removeAttr('disabled');
+    $('input#restricted_dirs').removeAttr('disabled');
+}
+
 function select_custom(){
     hide_all($create_datastore_dialog);
     $('select#ds_mad').val('fs');
