--- remotes/vmm/kvm/prereconfigure.orig	2022-01-06 14:54:03.880955380 +0200
+++ remotes/vmm/kvm/prereconfigure	2022-01-06 14:54:23.403149865 +0200
@@ -24,7 +24,7 @@
 
 if [[ -n "$DOMAIN" ]] && [[ -n "$TARGET_DEVICE" ]]
 then
-    disk="$(virsh --connect $LIBVIRT_URI domblklist $DOMAIN | awk "/^$TARGET_DEVICE/ {print \$2; exit}")"
+    disk="$(virsh --connect $LIBVIRT_URI domblklist $DOMAIN | awk "/^\s*$TARGET_DEVICE/ {print \$2; exit}")"
     if [[ "$disk" = "-" ]]; then
         log_debug "CDROM already ejected"
         exit 0
