diff -urN mads.orig/one_hm.rb mads/one_hm.rb
--- mads.orig/one_hm.rb	2019-05-20 14:12:31.000000000 +0300
+++ mads/one_hm.rb	2019-05-26 00:18:52.658614335 +0300
@@ -45,12 +45,24 @@
 
     def action_execute(number, hook_name, host, script, *arguments)
         cmd=nil
-        cmd_string="#{script} #{arguments.join(' ')}"
+        if arguments[0][0]=="-i"
+            cmd_string="#{script}"
+        else
+            cmd_string="#{script} #{arguments.join(' ')}"
+        end
 
         if host.upcase=="LOCAL"
-            cmd=LocalCommand.run(cmd_string, log_method(number))
+            if arguments[0][0]=="-i"
+                cmd=LocalCommand.run(cmd_string, log_method(number), arguments[0][1])
+            else
+                cmd=LocalCommand.run(cmd_string, log_method(number))
+            end
         else
-            cmd=SSHCommand.run("'#{cmd_string}'", host, log_method(number))
+            if arguments[0][0]=="-i"
+                cmd=SSHCommand.run("'#{cmd_string}'", host, log_method(number), arguments[0][1])
+            else
+                cmd=SSHCommand.run("'#{cmd_string}'", host, log_method(number))
+            end
         end
 
         if cmd.code==0
