diff -u mads.orig/one_tm.rb mads/one_tm.rb
--- mads.orig/one_tm.rb	2019-05-20 14:12:31.000000000 +0300
+++ mads/one_tm.rb	2019-05-26 10:44:24.949455260 +0300
@@ -99,7 +99,13 @@
     def do_transfer_action(id, command)
         cmd  = command[0].downcase
         tm   = command[1]
-        args = command[2..-1].map{|e| Shellwords.escape(e)}.join(" ")
+        if cmd.match(/(pre|post|fail)migrate/).nil?
+            stdin = nil
+            args = command[2..-1].map{|e| Shellwords.escape(e)}.join(" ")
+        else
+            args = command[2..-2].map{|e| Shellwords.escape(e)}.join(" ")
+            stdin = command[-1]
+        end
 
         if not @types.include?(tm)
             return RESULT[:failure], "Transfer Driver '#{tm}' not available"
@@ -107,7 +113,11 @@
 
         path = File.join(@local_scripts_path, tm, cmd)
         path << " " << args
-        rc = LocalCommand.run(path, log_method(id))
+        if stdin.nil?
+            rc = LocalCommand.run(path, log_method(id))
+        else
+            rc = LocalCommand.run(path, log_method(id), stdin)
+        end
 
         result, info = get_info_from_execution(rc)
 
