--- remotes/vmm/kvm/reconfigure.orig	2020-05-07 13:25:07.000000000 +0300
+++ remotes/vmm/kvm/reconfigure	2022-02-22 17:49:16.344492541 +0200
@@ -28,6 +28,46 @@
     CMD="virsh --connect $LIBVIRT_URI \
         change-media $DOMAIN $TARGET_DEVICE $ISO_PATH --insert"
 
+    CDROM_TYPE=$(stat -L --format=%F "$ISO_PATH" | awk '{print $1}')
+    [ "$CDROM_TYPE" != "block" ] || CMD="$CMD --block"
+
+    while read -u 4 qdev tray_open inserted_file locked; do
+        logger -t "vmm_sp_rec" -- "$qdev locked:$locked tray_open:$tray_open file:$inserted_file //loop"
+        if [ "${inserted_file##*/}" == "${ISO_PATH##*/}" ]; then
+            virsh --connect $LIBVIRT_URI qemu-monitor-command "$DOMAIN" --cmd "{\"execute\":\"eject\",\"arguments\":{\"id\":\"$qdev\"}}" &>"/tmp/json-rec-${DOMAIN}-eject"
+            logger -t "vmm_sp_rec" -- "qemu-monitor-command $DOMAIN --cmd '{\"execute\":\"eject\",\"arguments\":{\"id\":\"$qdev\"}}' ($?)"
+            virsh --connect $LIBVIRT_URI qemu-monitor-command "$DOMAIN" --cmd "{\"execute\":\"blockdev-open-tray\",\"arguments\":{\"id\":\"$qdev\"}}" &>"/tmp/json-rec-${DOMAIN}-open-tray"
+            logger -t "vmm_sp_rec" -- "qemu-monitor-command $DOMAIN --cmd {\"execute\":\"blockdev-open-tray\",\"arguments\":{\"id\":\"$qdev\"}} ($?)"
+            devid="$qdev"
+        fi
+	if [ "$inserted_file" == "null" ] && [ "$tray_open" == "false" ]; then
+            virsh --connect $LIBVIRT_URI qemu-monitor-command "$DOMAIN" --cmd "{\"execute\":\"blockdev-open-tray\",\"arguments\":{\"id\":\"$qdev\"}}" &>"/tmp/json-rec${DOMAIN}-open-tray"
+            logger -t "vmm_sp_rec" -- "qemu-monitor-command $DOMAIN --cmd {\"execute\":\"blockdev-open-tray\",\"arguments\":{\"id\":\"$qdev\"}} ($?)"
+            devid="$qdev"
+        fi
+    done 4< <(virsh --connect $LIBVIRT_URI \
+        qemu-monitor-command "$DOMAIN" --cmd '{"execute":"query-block"}' | tee "/tmp/json-rec-${DOMAIN}-query" |\
+        jq -r '.return[]|select(.removable)|.qdev+" "+(.tray_open|tostring)+" "+(.inserted.file|tostring)+" "+(.locked|tostring)' || true)
+
+    cnt=0
+    [ -z "$devid" ] || while : ; do
+    	tray_open=$(virsh --connect $LIBVIRT_URI qemu-monitor-command "$DOMAIN" --cmd '{"execute":"query-block"}' |\
+		        jq -r --arg qdev "$devid" '.return[]|select(.qdev==$qdev)|.tray_open')
+        logger -t "vmm_sp_pre" -- "$devid tray_open:$tray_open  //wait $((cnt*100))ms"
+	cnt=$((cnt + 1))
+	if [ "$tray_open" == "true" ] || [ $cnt -gt 20 ]; then
+            break
+        fi
+	sleep 0.1
+    done
+
+    logger -t "vmm_sp_rec" -- "$CMD # $CDROM_TYPE"
+
     exec_and_log "$CMD" "Could not insert CDROM $ISO_PATH to $TARGET_DEVICE"
+
+    while read -u 4 q t f l; do
+        logger -t "vmm_sp_rec" -- "$q locked:$l tray_open:$t file:$f //END"
+    done 4< <(virsh --connect $LIBVIRT_URI qemu-monitor-command "$DOMAIN" --cmd '{"execute":"query-block"}' |\
+        jq -r '.return[]|select(.removable)|.qdev+" "+(.tray_open|tostring)+" "+(.inserted.file|tostring)+" "+(.locked|tostring)')
 fi
 

